src/lib/Name.hs

```diff
@@ -2600,30 +2576,39 @@ instance (ExtEnd n => ExtEnd l) => Ext n l
 class ExtEnd (n::S)
 
 getExtEvidence :: Ext n l => ExtEvidence n l
-getExtEvidence = FabricateExtEvidence
+getExtEvidence = ExtEvidence
 
 absurdExtEvidence :: ExtEvidence VoidS n
-absurdExtEvidence = FabricateExtEvidence
+absurdExtEvidence = fabricateExtEvidence
 
 -- We give this one a ' because the more general one defined in Name is the
 -- version we usually want to use.
 withExtEvidence' :: forall n l a. ExtEvidence n l -> (Ext n l => a) -> a
-withExtEvidence' _ cont = fromWrapWithExt
+withExtEvidence' ExtEvidence cont = cont
+{-# INLINE withExtEvidence' #-}
+
+withFabricatedExt :: forall n l a. (Ext n l => a) -> a
+withFabricatedExt cont = fromWrapWithExt
  ( TrulyUnsafe.unsafeCoerce ( WrapWithExt cont :: WrapWithExt n     l     a
                                              ) :: WrapWithExt VoidS VoidS a)
-{-# INLINE withExtEvidence' #-}
+{-# INLINE withFabricatedExt #-}
+
+fabricateExtEvidence :: forall n l. ExtEvidence n l
+fabricateExtEvidence = withFabricatedExt @n @l ExtEvidence
+{-# INLINE fabricateExtEvidence #-}
 
 newtype WrapWithExt n l r =
   WrapWithExt { fromWrapWithExt :: Ext n l => r }
 
-data ExtEvidence (n::S) (l::S) = FabricateExtEvidence
+data ExtEvidence (n::S) (l::S) where
+  ExtEvidence :: (Ext n l) => ExtEvidence n l
 
 instance Category ExtEvidence where
-  id = FabricateExtEvidence
+  id = ExtEvidence
   {-# INLINE id #-}
   -- Unfortunately, we can't write the class version of this transitivity axiom
   -- because the intermediate type would be ambiguous.
```

Comment:
I feel like this comment is false? We have the trick that allows GHC to infer transitivity

Reply:
Well, it's certainly quite old, so it could be stale.  Removed.